File: app/src/main/java/cn/ftoutiao/account/android/base/impl/IActionBarResIdProvider.java
package cn.ftoutiao.account.android.base.impl;

import android.support.annotation.IdRes;

/**
 * Created by yangweichao on 2018/3/7.
 */

public interface IActionBarResIdProvider {

    @IdRes
    int getActionBarId();

    @IdRes
    int getActionBarTitleId();

    @IdRes
    int getActionBarContainerId();

    @IdRes
    int getActionBarIconId();

    @IdRes
    int getActionBarActionId();
}

File: ComponentLib/src/main/java/cn/ftoutiao/account/android/component/widget/IActionBarResIdProvider.kt
package cn.ftoutiao.account.android.component.widget

import android.support.annotation.IdRes

/**
 * Created by alan on 2017/11/4.
 */
interface IActionBarResIdProvider {

    @IdRes abstract fun getActionBarId(): Int

    @IdRes abstract fun getActionBarTitleId(): Int

    @IdRes abstract fun getActionBarContainerId(): Int

    @IdRes abstract fun getActionBarIconId(): Int

    @IdRes abstract fun getActionBarActionId(): Int
}